src/textwin.cpp

```diff
@@ -219,15 +153,13 @@ static ShowHideButton pointsButton =
     { &(SS.GW.showPoints),       "point",         "points"                          };
 static ShowHideButton constructionButton =
     { &(SS.GW.showConstruction), "construction",  "construction entities"           };
-static TriStateButton constraintsButton =
-    { &(SS.GW.showConstraints),
-      {GraphicsWindow::ShowConstraintMode::SCM_SHOW_ALL,
-       GraphicsWindow::ShowConstraintMode::SCM_SHOW_DIM,
-       GraphicsWindow::ShowConstraintMode::SCM_NOSHOW},
-      {"constraints and dimensions","dimensions","none"},
-      {"constraint","constraint-dimo","constraint-wo"}
-    };
-
+static TriStateButton constraintsButton  = {
+    (unsigned *)(&(SS.GW.showConstraints)),
+    {(unsigned)GraphicsWindow::ShowConstraintMode::SCM_SHOW_ALL,
+     (unsigned)GraphicsWindow::ShowConstraintMode::SCM_SHOW_DIM,
+     (unsigned)GraphicsWindow::ShowConstraintMode::SCM_NOSHOW},
+    {"Show only dimensions", "Hide constraints and dimensions", "Show constraints and dimensions"},
```

Comment:
{"Show constraints and dimensions", "Show only dimensions", "Hide constraints and dimensions"}

Reply:
@77maxikov The idea is that the tool tip should show what will happen when the user clicks the button instead of the current state. Take a look at how the hidden line one worked even before.

Reply:
Ok, got it.